client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -80,6 +82,12 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
         }
     }
 
+    public RawClient(PravegaNodeUri uri, ConnectionFactory connectionFactory) {
```

Comment:
Before this change the RawClient was always tied to specific segment. With this change we are making a RawClient specific to a `PravegaNodeUri`. This implies a given RawClient can be used to perform operations on all the segments owned by a specific Segment Store.  (Hence we could have a pool of RawClient instances where there is atleast one instance per Segment Store. )

Also, `connectionFactory.establishConnection(uri, responseProcessor);` in line 87 below internally establishes a new connection , `establishConnection()` is invoked by this method and it disables Flow/ConnectionPooling.

Hence we should create a connection using `connectionFactory.establishConnection(flow, uri, responseProcessor)` similar to the segment specific RawClient constructor.